src/commands.ts

```diff
@@ -503,12 +504,16 @@ export class Commands {
 						this.restClient,
 						toSafeHost(url),
 					);
-					const escape = (str: string): string =>
-						`"${str.replace(/"/g, '\\"')}"`;
+
+					const configDir = path.dirname(
+						this.storage.getSessionTokenPath(toSafeHost(url)),
+					);
+					const globalFlags = getGlobalFlags(
+						vscode.workspace.getConfiguration(),
+						configDir,
+					);
 					terminal.sendText(
-						`${escape(binary)} ssh --global-config ${escape(
-							path.dirname(this.storage.getSessionTokenPath(toSafeHost(url))),
-						)} ${app.workspace_name}`,
+						`${escapeCommandArg(binary)}${globalFlags.length === 0 ? "" : ` ${globalFlags.join(" ")}`} ssh ${app.workspace_name}`,
```

Comment:
Now that we know there is always a `--global-config` flag at least, we could skip the length check.